### ISSUE TITLE
Error with agent.setConfiguration()

### ISSUE THREAD

**Issue by Juzziee** [amazon-connect/amazon-connect-streams#460]
It appears that the latest version of the amazon-connect-streams has an issue when trying to set agent configuration. 

when agent.setConfiguration() is called, the following command is run;

` Agent.prototype.setConfiguration = function (configuration, callbacks) {
    var client = connect.core.getClient();
    if (configuration && configuration.agentPreferences && !connect.isValidLocale(configuration.agentPreferences.locale)) {
      if (callbacks && callbacks.failure) {
        callbacks.failure(connect.AgentErrorStates.INVALID_LOCALE);
      }
    } else {
      client.call(connect.ClientMethods.UPDATE_AGENT_CONFIGURATION, {
        configuration: connect.assertNotNull(configuration, 'configuration')
      }, {
          success: function (data) {
            // We need to ask the shared worker to reload agent config
            // once we change it so every tab has accurate config.
            var conduit = connect.core.getUpstream();
            conduit.sendUpstream(connect.EventType.RELOAD_AGENT_CONFIGURATION);

            if (callbacks.success) {
              callbacks.success(data);
            }
          },
          failure: callbacks && callbacks.failure
      });
    }
  }; `

The line in question is
`if (configuration && configuration.agentPreferences && !connect.isValidLocale(configuration.agentPreferences.locale)) {`

The agent config refers to the locale as `configuration.agentPreferences.LANGAUGE`

For anyone else experiencing errors with their CCP, just update this line to read
`if (configuration && configuration.agentPreferences && !connect.isValidLocale(configuration.agentPreferences.LANGUAGE)) {`

 Thanks :)

**Comment by seiyako** [comment]
Hi @Juzziee,

Thanks for reporting the issue, this is apparently a bug. While we're working on fixing this bug, a workaround you can try is to set a `locale` field as well to the configuration object you are passing in.
```javascript
const config = agent.getConfiguration();
config.locale = config.agentPreference.LANGUAGE;
agent.sestConfiguration(config);
```

**Comment by alexgurr** [comment]
The code snippet above from @seiyako is incorrect, it's `agentPreferences` not `agentPreference` as suggested and doesn't actually work, we still couldn't get past this error, even with the proposed fix. Will try @Juzziee 's suggestion and fork inside node modules.

**Comment by jeve7** [comment]
Hi Team, do we know when is this going to be fixed? I was testing and the workaround worked for me. BTW @alexgurr is right right, it's agentPreferences.

```var config = agent.getConfiguration(); 
config.softphoneEnabled = true; // true or false
config.extension = ""; // Extension number if softphoneEnabled is false. Must be E.164.
// Bug workaround: Set the locale from "LANGUAGE" to "locale"
config.agentPreferences.locale = config.agentPreferences.LANGUAGE;
agent.setConfiguration(config, {
    success: function() { 
        console.info("agent.setConfiguration worked");
    },
    failure: function(err) { 
        console.error("agent.setConfiguration FAILED", err);
     },
   }
);```

**Comment by seiyako** [comment]
The property inconsistency was fixed at v2.2.0. The following code should work now (tested in browser's console).

```
let agent = new connect.Agent();
let config = agent.getConfiguration();
config.agentPreferences.LANGUAGE = 'en_US';
agent.setConfiguration(config, { success: () => console.log('success'), failure: () => console.log('error;') });
```
